chat_server/index.ts

```diff
@@ -1,52 +1,73 @@
-import * as express from "express";
-import socketioJwt from 'socketio-jwt';
-import socketIo from 'socket.io';
-import { createServer, Server, request } from 'http';
-import {add, get} from './db';
+const express = require("express");
+const socketio_jwt = require('socketio-jwt');
+import { createServer} from 'http';
+
+const dotenv = require('dotenv')
+dotenv.config()
+
+const db = require('./db/db');
+
 
 class Message {
-	constructor(public user_id: string, public message: string, posted_on: Date){}
+	constructor(public username: string, public message: string, public posted_on: Date){
+        this.username = username;
+        this.message = message;
+        this.posted_on = posted_on;
+    }
 }
 
 /**
  * The JWT authentication is made with https://github.com/auth0-community/auth0-socketio-jwt
  */
 
 const app = express();
-app.set("port", process.env.PORT || 3000);
+var port = process.env.PORT || 3000
+
+export let httpServer = createServer(app);
+export let io = require('socket.io')(httpServer);
+
+// Authentification using handshake
+var jwt_option = {
+    "secret": process.env.JWT_SECRET,
+    "timeout": 1000,
+    "handshake": true
+};
 
-let httpServer = createServer(app);
-let io = socketIo(httpServer);
+io.use(socketio_jwt.authorize(jwt_option));
 
-io
-.on('connection', socketioJwt.authorize({
-	secret: 'your secret or public key',
-	handshake: true,
-	decodedPropertyName: ""
-}))
-.on("authenticated", (socket: any) => {
+io.sockets
+.on('connection', (socket) => {
 
-	socket.on("message", (request: any) => {
+	socket.on("message", async (request: any) => {
 		if (request.message === undefined ){
```

Comment:
```suggestion
		if (!request.message){
```
Should also filtre empty messages to check)